src/PersistentPageIdentifiersExtension.php

```diff
@@ -29,11 +30,21 @@ public function getIdGenerator(): IdGenerator {
 	}
 
 	public function getPersistentPageIdentifiersRepo(): PersistentPageIdentifiersRepo {
+		if ( defined( 'MW_PARSER_TEST' ) ) {
+			return $this->getParserTestPersistentPageIdentifiersRepo();
+		}
+
 		return new DatabasePersistentPageIdentifiersRepo(
 			$this->getDatabase()
 		);
 	}
 
+	private function getParserTestPersistentPageIdentifiersRepo(): PersistentPageIdentifiersRepo {
+		return new StubPersistentPageIdentifiersRepo(
+			$GLOBALS['wgPersistentPageIdentifiersParserTestStubId'] ?? null
```

Comment:
Not using `MediaWikiServices::getInstance()->getMainConfig()->get()` because then we have to declare this in `extension.json`. This serves no production purpose, so no need to pollute allowed variables.